lib/index.js

```diff
@@ -0,0 +1,96 @@
+const config = require('./config');
+
+var request = require('request-promise-native');
+
+
+module.exports = function(apikey) {
+  this.key = apikey;
+
+  this.samlEntity = function(name, entityId, federationId) {
```

Comment:
I would consider moving the models do a different file, IdentityProvider, or something from the endpoints